libnopegl/src/distmap.c

```diff
@@ -0,0 +1,741 @@
+
+    s->max_shape_w = NGLI_MAX(shape_w, s->max_shape_w);
+    s->max_shape_h = NGLI_MAX(shape_h, s->max_shape_h);
+
+    *shape_id = (int32_t)ngli_darray_count(&s->shapes) - 1;
+    return 0;
+}
+
+static int32_t get_beziergroup_start(const struct distmap *s, int32_t shape_id)
+{
+    const int32_t *group_counts = ngli_darray_data(&s->beziergroup_counts);
+    int32_t group_count = 0;
+    for (int32_t i = 0; i < shape_id; i++)
+        group_count += group_counts[i];
+    return group_count;
+}
+
+static int32_t get_beziergroup_end(const struct distmap *s, int32_t shape_id)
+{
+    const int32_t group_start = get_beziergroup_start(s, shape_id);
+    const int32_t *group_counts = ngli_darray_data(&s->beziergroup_counts);
+    return group_start + group_counts[shape_id];
+}
+
+static int32_t get_bezier_start(const struct distmap *s, int32_t shape_id)
+{
+    const int32_t group_start = get_beziergroup_start(s, shape_id);
+    const int32_t *counts = ngli_darray_data(&s->bezier_counts);
+    int32_t start = 0;
+    for (int32_t i = 0; i < group_start; i++)
+        start += abs(counts[i]);
+    return start;
+}
+
+static int32_t get_bezier_end(const struct distmap *s, int32_t shape_id)
+{
+    const int32_t group_end = get_beziergroup_end(s, shape_id);
+    const int32_t *counts = ngli_darray_data(&s->bezier_counts);
+    int32_t end = 0;
+    for (int32_t i = 0; i < group_end; i++)
+        end += abs(counts[i]);
+    return end;
+}
+
+/*
+ * Get the maximum number of beziers across all shapes. This is useful to
+ * get how large the bezier uniform buffer must be (it will be re-used for
+ * each shape).
+ */
+static int32_t get_max_beziers_per_shape(const struct distmap *s)
+{
+    int32_t max_beziers = 0;
+    const int32_t *counts = ngli_darray_data(&s->bezier_counts);
+    const int32_t *group_counts = ngli_darray_data(&s->beziergroup_counts);
+    for (size_t i = 0; i < ngli_darray_count(&s->beziergroup_counts); i++) {
+        const int32_t group_count = group_counts[i];
+
+        int32_t sum = 0;
+        for (int32_t j = 0; j < group_count; j++)
+            sum += abs(counts[j]);
+        counts += group_count;
+
+        max_beziers = NGLI_MAX(max_beziers, sum);
+    }
+    return max_beziers;
+}
+
+static int32_t get_max_beziergroups_per_shape(const struct distmap *s)
+{
+    int32_t max_groups = 0;
+    const int32_t *group_counts = ngli_darray_data(&s->beziergroup_counts);
+    for (size_t i = 0; i < ngli_darray_count(&s->beziergroup_counts); i++)
+        max_groups = NGLI_MAX(max_groups, group_counts[i]);
+    return max_groups;
+}
+
+enum {
+    VERTICES_INDEX,
+};
+
+enum {
+    COORDS_INDEX,
+    SCALE_INDEX,
+    BEZIER_X_BUF_INDEX,
+    BEZIER_Y_BUF_INDEX,
+    BEZIER_COUNTS_INDEX,
+    BEZIERGROUP_COUNT_INDEX,
+};
+
+static void load_buffers_data(struct distmap *s,
+                              uint8_t *vert_data, size_t vert_offset,
+                              uint8_t *frag_data, size_t frag_offset)
```

Comment:
`vert_offset` and `frag_offset` are unused (the function uses `s->vert_offset` and `s->frag_offset`.

Reply:
Mmh yeah, removed